app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3668,41 +3664,45 @@ private void setupSubtitleView() {
         binding.subtitleView.setStyle(captionStyle);
     }
 
-    private void onTextTracksChanged() {
-        final int textRenderer = getCaptionRendererIndex();
-
+    private void onTextTracksChanged(@NonNull final TracksInfo currentTrackInfo) {
         if (binding == null) {
             return;
         }
+
         if (trackSelector.getCurrentMappedTrackInfo() == null
-                || textRenderer == RENDERER_UNAVAILABLE) {
+                || !currentTrackInfo.isTypeSupportedOrEmpty(C.TRACK_TYPE_TEXT)) {
             binding.captionTextView.setVisibility(View.GONE);
             return;
         }
 
-        final TrackGroupArray textTracks = trackSelector.getCurrentMappedTrackInfo()
-                .getTrackGroups(textRenderer);
-
         // Extract all loaded languages
-        final List<String> availableLanguages = new ArrayList<>(textTracks.length);
-        for (int i = 0; i < textTracks.length; i++) {
-            final TrackGroup textTrack = textTracks.get(i);
+        final List<TracksInfo.TrackGroupInfo> textTracks = currentTrackInfo
+                .getTrackGroupInfos()
+                .stream()
+                .filter(trackGroupInfo -> C.TRACK_TYPE_TEXT == trackGroupInfo.getTrackType())
+                .collect(Collectors.toList());
+        final List<String> availableLanguages = new ArrayList<>(textTracks.size());
+        for (int i = 0; i < textTracks.size(); i++) {
+            final TrackGroup textTrack = textTracks.get(i).getTrackGroup();
             if (textTrack.length > 0) {
                 availableLanguages.add(textTrack.getFormat(0).language);
             }
         }
```

Comment:
I think this can be converted to java-streams api:

Example:
```java
...
final List<String> availableLanguages = textTracks 
  .filter(info -> info.getTrackGroup().length > 0) // Maybe use isEmpty if available
  .map(textTrack -> textTrack.getFormat(0))
  .collect(Collectors.toList())
```

Reply:
Thanks, fixed.